python/lsst/afw/geom/Extent.i

```diff
@@ -47,12 +46,138 @@
+    lsst::afw::geom::Point<T,N> __add__(lsst::afw::geom::Point<T,N> const & other) const {
+        return *self + other;
+    }
+    lsst::afw::geom::Extent<T,N> __add__(lsst::afw::geom::Extent<T,N> const & other) const {
+        return *self + other;
+    }
+    lsst::afw::geom::Extent<T,N> __sub__(lsst::afw::geom::Extent<T,N> const & other) const {
+        return *self - other;
+    }
+    PyObject * __iadd__(PyObject** PYTHON_SELF, lsst::afw::geom::Extent<T,N> const & other) {
+        *self += other;
+        Py_INCREF(*PYTHON_SELF);
+        return *PYTHON_SELF;
+    }
+    PyObject * __isub__(PyObject** PYTHON_SELF, lsst::afw::geom::Extent<T,N> const & other) {
+        *self -= other;
+        Py_INCREF(*PYTHON_SELF);
+        return *PYTHON_SELF;
+    }
+    lsst::afw::geom::Extent<T,N> __mul__(T scalar) const {
+        return *self * scalar;
+    }
+    PyObject * __imul__(PyObject** PYTHON_SELF, T scalar) {
+        *self *= scalar;
+        Py_INCREF(*PYTHON_SELF);
+        return *PYTHON_SELF;
+    }
+    lsst::afw::geom::Extent<T,N> __rmul__(T scalar) const {
+        return scalar * (*self);
+    }
+}
+%enddef
+
+%define %ExtentD_POSTINCLUDE(N)
+%Extent_POSTINCLUDE(double, N, D)
+%extend lsst::afw::geom::Extent<double,N> {
+    lsst::afw::geom::Extent<int,N> truncate() const { return truncate(*self); }
+    lsst::afw::geom::Extent<int,N> floor() const { return floor(*self); }
+    lsst::afw::geom::Extent<int,N> ceil() const { return ceil(*self); }
+    lsst::afw::geom::Extent<double,N> __add__(lsst::afw::geom::Extent<int,N> const & other) const {
+        return *self + other;
+    }
+    lsst::afw::geom::Point<double,N> __add__(lsst::afw::geom::Point<int,N> const & other) const {
+        return other + *self;
+    }
+    PyObject * __iadd__(PyObject** PYTHON_SELF, lsst::afw::geom::Extent<int,N> const & other) {
+        *self += other;
+        Py_INCREF(*PYTHON_SELF);
+        return *PYTHON_SELF;
+    }
+    lsst::afw::geom::Extent<double,N> __sub__(lsst::afw::geom::Extent<int,N> const & other) const {
+        return *self - other;
+    }
+    PyObject * __isub__(PyObject** PYTHON_SELF, lsst::afw::geom::Extent<int,N> const & other) {
+        *self -= other;
+        Py_INCREF(*PYTHON_SELF);
+        return *PYTHON_SELF;
+    }
+    // no need for overloads that take int scalars, because int scalars will automatically match the
+    // overloads that take double
+    lsst::afw::geom::Extent<double,N> __truediv__(double scalar) const {
+        return *self / scalar;
+    }
+    PyObject * __itruediv__(PyObject** PYTHON_SELF, double scalar) {
+        *self /= scalar;
+        Py_INCREF(*PYTHON_SELF);
+        return *PYTHON_SELF;
+    }
     %pythoncode %{
 # support "__from__ future import division" in Python 2; not needed for Python 3
 # warning: this indentation level is required for SWIG 3.0.2, at least
-__truediv__ = __div__
-__itruediv__ = __idiv__
+__div__ = __truediv__
+__idiv__ = __itruediv__
```

Comment:
I think the comment here is misleading -- post changes, this is defining `__div__`, not `__truediv__`, so it's setting normal (rather than future) division.

Plus given that this is division for an `ExtentD`, I'm not sure why future division comes into it at all -- you're always expecting to get doubles back, right?